src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,361 @@
+              "Source directory contains live restore stop file: %s. This implies it is a "
+              "destination directory that hasn't finished restoration",
+              dirlist_source[i]);
+
+    /* Now check the destination folder */
+
+    /* Read state directly from the turtle file in the destination. */
+    WT_ERR(__live_restore_get_state_from_file(session, lr_fs->os_file_system, &state_from_file));
+
+    WT_ERR(lr_fs->os_file_system->fs_directory_list(lr_fs->os_file_system, (WT_SESSION *)session,
+      lr_fs->destination.home, "", &dirlist_dest, &num_dest_files));
+
+    switch (state_from_file) {
+    case WTI_LIVE_RESTORE_STATE_NONE:
+        /*
+         * Ideally we'd prevent live restore from starting when there are any files already present
+         * in the destination, but we can't control for everything that the user might put into the
+         * folder. Instead only check for WiredTiger files.
+         */
+        for (uint32_t i = 0; i < num_dest_files; ++i) {
+            if (WT_PREFIX_MATCH(dirlist_dest[i], WT_WIREDTIGER) ||
+              WT_SUFFIX_MATCH(dirlist_dest[i], ".wt"))
+                WT_ERR_MSG(session, EINVAL,
+                  "Attempting to begin a live restore on a directory that already contains "
+                  "WiredTiger files '%s'! It's possible this file will be overwritten.",
+                  dirlist_dest[i]);
+        }
+        break;
+    case WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION:
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        /* There's no invalid state to check in these cases. */
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        for (uint32_t i = 0; i < num_dest_files; ++i) {
+            if (WT_SUFFIX_MATCH(dirlist_dest[i], WTI_LIVE_RESTORE_STOP_FILE_SUFFIX))
+                WT_ERR_MSG(session, EINVAL,
+                  "Live restore is complete but live restore stop file '%s' still exists!",
+                  dirlist_dest[i]);
+        }
+        break;
+    }
+
+err:
+    WT_TRET(lr_fs->os_file_system->fs_directory_list_free(
+      lr_fs->os_file_system, (WT_SESSION *)session, dirlist_source, num_source_files));
+
+    WT_TRET(lr_fs->os_file_system->fs_directory_list_free(
+      lr_fs->os_file_system, (WT_SESSION *)session, dirlist_dest, num_dest_files));
+
+    return (ret);
+}
+
+/*
+ * __wt_live_restore_validate_non_lr_system --
+ *     When starting in non-live restore mode make sure we're not in the middle of a live restore.
+ */
+int
+__wt_live_restore_validate_non_lr_system(WT_SESSION_IMPL *session)
+{
+    /* The turtle file should indicate we're not in the middle of a live restore. */
+    WTI_LIVE_RESTORE_STATE state;
+    __live_restore_get_state_from_file(session, S2C(session)->file_system, &state);
+    if (state != WTI_LIVE_RESTORE_STATE_NONE && state != WTI_LIVE_RESTORE_STATE_COMPLETE)
+        WT_RET_MSG(session, EINVAL,
+          "Cannot start in non-live restore mode while a live restore is in progress!");
+
```

Comment:
We could add some logic to abort if we find stop files in the source directory, but I'm not convinced it's worth the additional lines of code. I have it locally if we want to add this check